src/rust/ide/src/model/synchronized/graph/executed.rs

```diff
@@ -82,17 +122,67 @@ impl Handle {
     /// context.
     pub fn subscribe(&self) -> impl Stream<Item=Notification> {
         let registry     = self.execution_ctx.computed_value_info_registry();
-        let value_stream = registry.subscribe().map(Notification::ComputedValueInfo);
-        let graph_stream = self.graph.subscribe().map(Notification::Graph);
-        futures::stream::select(value_stream,graph_stream)
+        let value_stream = registry.subscribe().map(Notification::ComputedValueInfo).boxed_local();
+        let graph_stream = self.graph().subscribe().map(Notification::Graph).boxed_local();
+        let self_stream = self.notifier.borrow_mut().subscribe().boxed_local();
+        futures::stream::select_all(vec![value_stream,graph_stream,self_stream])
     }
-}
 
-impl Deref for Handle {
-    type Target = controller::Graph;
+    /// Create a graph controller for the given method.
+    ///
+    /// Fails if the module is inaccessible or if it does not contain given method.
+    pub async fn graph_for_method
+    (&self, method:&MethodPointer) -> FallibleResult<controller::Graph> {
+        let module_path = model::module::Path::from_file_path(method.file.clone())?;
+        let module      = self.project.module_controller(module_path).await?;
+        debug!(self.logger,"Looking up method definition {method:?} in the module.");
+        let module_ast = module.model.model.ast();
+        let definition = double_representation::module::lookup_method(&module_ast,method)?;
+        module.graph_controller(definition)
+    }
+
+    /// Step into node by given ID.
+    ///
+    /// This will cause pushing a new stack frame to the execution context and changing the graph
+    /// controller to point to a new definition.
+    ///
+    /// Fails if there's no information about target method pointer (e.g. because node value hasn't
+    /// been yet computed by the engine) or if method graph cannot be created (see
+    /// `graph_for_method` documentation).
+    pub async fn step_into_node(&self, node:double_representation::node::Id) -> FallibleResult<()> {
+        debug!(self.logger, "Entering node {node}");
+        let registry   = self.execution_ctx.computed_value_info_registry();
+        let node_info  = registry.get(&node).ok_or_else(|| NotEvaluatedYet(node))?;
+        let method_ptr = node_info.method_call.as_ref().ok_or_else(|| NoResolvedMethod(node))?;
+        let graph      = self.graph_for_method(method_ptr).await?;
+        let call       = model::execution_context::LocalCall {
+            call : node,
+            definition : method_ptr.clone()
+        };
```

Comment:
why not `let call       = model::execution_context::LocalCall::new(node,method_ptr)` instead?

Reply:
I don't think new would improve this.